runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterOverlay.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2018, Zimaya <https://github.com/Zimaya>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.regenmeter;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.geom.Rectangle2D;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+public class RegenMeterOverlay extends Overlay
+{
+	private final Client client;
+	private RegenTimer timer;
+
+	@Inject
+	public RegenMeterOverlay(Client client)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		timer = new RegenTimer(client);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point point)
+	{
+		Widget hpOrb = client.getWidget(WidgetInfo.MINIMAP_HEALTH_ORB);
+		if (hpOrb == null)
+		{
+			return null;
+		}
+
+		Rectangle2D bounds = hpOrb.getBounds().getBounds2D();
+		if (bounds.getX() <= 0)
+		{
+			return null;
+		}
+
+		int orbInnerX = (int) (bounds.getX() + 27);
+		int orbInnerY = (int) (bounds.getY() + 4);
+
+		graphics.setColor(new Color(255, 255, 255, 100));
+		graphics.fillArc(orbInnerX, orbInnerY, 26, 26, 90, (int) (3.6f * timer.getProgress()));
```

Comment:
as @Abextm stated we will want to either cache this fillArc call or maybe draw the arc ourselves. I'm not sure the best way forward on this.